model_analyzer/config/run/model_run_config.py

```diff
@@ -115,6 +115,38 @@ def is_legal_combination(self):
             logger.debug(
                 f"Illegal model run config because client batch size {perf_batch_size} is greater than model max batch size {max_batch_size}"
             )
+
+        return legal
+
+    def check_for_preferred_vs_model_batch_size(self):
+        """
+        Returns false if maximum of preferred batch size is greater than model batch size. Else true
+        """
+        model_config = self._model_config.get_config()
+        legal = True
+
+        max_batch_size = model_config[
+            'max_batch_size'] if 'max_batch_size' in model_config else self.DEFAULT_MAX_BATCH_SIZE
+
+        if 'dynamic_batching' in model_config and 'preferred_batch_size' in model_config['dynamic_batching']:
+            max_preferred_batch_size = max(model_config['dynamic_batching']['preferred_batch_size'])
+            legal = max_batch_size >= max_preferred_batch_size
+
+            if not legal:
+                logger.debug(
+                    f"Illegal model run config because maximum of model preferred batch size {max_preferred_batch_size} is greater than model max batch size {max_batch_size}"
+            )
+
+        return legal
+    
+    def is_legal_combination(self):
+        """
+        Returns true if the run_config is valid and should be run. Else false
+        """
+        legal = self.check_for_client_vs_model_batch_size()
```

Comment:
I think this would be cleaner with the if statement. Along the lines of: ```legal = self.check_for_client_vs_model_batch_size() and self.check_for_client_vs_model_batch_size()```

Reply:
Yes, done the modification 👍